### ISSUE TITLE
VICReg requires torch >= 1.10

### ISSUE THREAD

**Issue by PDomonkos** [KevinMusgrave/pytorch-metric-learning#467]
Hi,

The torch.cov in the covariance_loss function requires torch >= 1.10.
Replacing:
>      _, D = emb.size()
>      cov_emb = torch.cov(emb.T)
>      cov_ref_emb = torch.cov(ref_emb.T)
to:
>      batch_size, D = emb.size()
>      cov_emb = (emb.T @ emb) / (batch_size - 1)
>      cov_ref_emb = (ref_emb.T @ ref_emb) / (batch_size - 1)
would make it compatible with older versions as well.

Another remark is that in the reference implementation, the variance loss is divided by 2: https://github.com/facebookresearch/vicreg/blob/5e7b38f4586384bbb0d9a035352fab1d8f03b3b4/main_vicreg.py#L207
Was it intentionally left out, or am I missing something here?

**Comment by KevinMusgrave** [comment]
Thanks for the tip regarding pytorch version.

Re: division by 2, I don't think it was left out intentionally.

**Comment by cwkeam** [comment]
@KevinMusgrave Hi! Actually I'll take a look at this since I contributed to this loss function. It's a bit late in my time zone right now so I'll take a look tomorrow.

**Comment by KevinMusgrave** [comment]
Thanks @cwkeam!

**Comment by KevinMusgrave** [comment]
Fixed in v1.3.1
